junit5/src/main/java/cz/xtf/junit5/extensions/OpenShiftRecorderService.java

```diff
@@ -0,0 +1,527 @@
+    private static final String CONFIGMAP_FILTER_MASTER = "CONFIGMAP_FILTER_MASTER";
+    private static final String SERVICE_FILTER_MASTER = "SERVICE_FILTER_MASTER";
+    private static final String EVENT_FILTER_MASTER = "EVENT_FILTER_MASTER";
+    private static final String POD_FILTER_BUILDS = "POD_FILTER_BUILDS";
+    private static final String BUILD_FILTER_BUILDS = "BUILD_METHOD_FILTER_BUILDS";
+    private static final String BC_FILTER_BUILDS = "BC_FILTER_BUILDS";
+    private static final String IS_FILTER_BUILDS = "IS_FILTER_BUILDS";
+    private static final String EVENT_FILTER_BUILDS = "EVENT_FILTER_BUILDS";
+
+    /**
+     * Initialize filters
+     */
+    public void initFilters(ExtensionContext context) {
+        ExtensionContext.Store classStore = getClassStore(context);
+        OpenShift master = OpenShifts.master();
+        OpenShift bm = BuildManagers.get().openShift();
+
+        initClassFilter(context, POD_FILTER_MASTER, master, Pod.class);
+        initClassFilter(context, DC_FILTER_MASTER, master, DeploymentConfig.class);
+        initClassFilter(context, BUILD_FILTER_MASTER, master, Build.class);
+        initClassFilter(context, BC_FILTER_MASTER, master, BuildConfig.class);
+        initClassFilter(context, IS_FILTER_MASTER, master, ImageStream.class);
+        initClassFilter(context, SS_FILTER_MASTER, master, StatefulSet.class);
+        initClassFilter(context, ROUTE_FILTER_MASTER, master, Route.class);
+        initClassFilter(context, CONFIGMAP_FILTER_MASTER, master, ConfigMap.class);
+        initClassFilter(context, SERVICE_FILTER_MASTER, master, Service.class);
+        classStore.put(EVENT_FILTER_MASTER,
+                new EventsFilterBuilder().setExcludedUntil(ResourcesTimestampHelper.timeOfLastEvent(master)));
+
+        // builds namespace (if not same)
+        if (!isMasterAndBuildNamespaceSame()) {
+            initClassFilter(context, POD_FILTER_BUILDS, bm, Pod.class);
+            initClassFilter(context, BUILD_FILTER_BUILDS, bm, Build.class);
+            initClassFilter(context, BC_FILTER_BUILDS, bm, BuildConfig.class);
+            initClassFilter(context, IS_FILTER_BUILDS, bm, ImageStream.class);
+            classStore.put(EVENT_FILTER_BUILDS,
+                    new EventsFilterBuilder().setExcludedUntil(ResourcesTimestampHelper.timeOfLastEvent(bm)));
+        }
+
+        classStore.put(FILTER_FOR_BEFORE_ALL_DONE, new AtomicBoolean(false));
+    }
+
+    /**
+     * Reset filters
+     */
+    public void resetFilters(ExtensionContext context) {
+        ExtensionContext.Store classStore = getClassStore(context);
+        OpenShift master = OpenShifts.master();
+        OpenShift bm = BuildManagers.get().openShift();
+        if (!classStore.get(FILTER_FOR_BEFORE_ALL_DONE, AtomicBoolean.class).get()) {
+            // so far filters have filtered everything until _BEFORE_ALL_ - TIME_1
+            // now we are before _BEFORE_EACH_ and implicitly _BEFORE_ALL_ is successful
```

Comment:
we are in the moment.

Reply:
Sorry but I don't understand. Please elaborate.

Reply:
if you meant: s/now we are before/now we are in the moment before/ then I have changed the comment by removing  lifecycle related aspects